cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -120,12 +120,51 @@ struct InsnOutput {
     output: usize,
 }
 
-fn input_to_reg<'a>(ctx: Ctx<'a>, spec: InsnInput) -> Reg {
+fn input_to_reg(ctx: Ctx, spec: InsnInput) -> Reg {
     let inputs = ctx.get_input(spec.insn, spec.input);
     ctx.use_input_reg(inputs);
     inputs.reg
 }
 
+enum ExtSpec {
+    ZeroExtend32,
```

Comment:
nit-level, at best, but .. (eg) `*Extend32` doesn't make it obvious to the casual reader whether the 32 is the source or dest width.  Would you consider adding `To` in front of the number?  eg `*ExtendTo32` ?